### PR TITLE
port ngx_http_lua_api.c to ngx_stream_lua_api.c

### PR DESCRIPTION
**Pull request by rainingmaster**
this pr is ported directly from ngx_http_lua_module's [feature](https://github.com/openresty/lua-nginx-module/commit/73bd64dbc145bc89c049075532ae841f7d7b6e02)

thanks for @doujiang24 's help : )

### CODE REVIEW

**Comment by rainingmaster** [comment]
[The error](https://travis-ci.org/rainingmaster/stream-lua-nginx-module/jobs/183166865#L805) maybe we can fix it by [this pr](https://github.com/openresty/stream-lua-nginx-module/pull/56)

File: src/ngx_stream_lua_api.c [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+ngx_shm_zone_t *
+ngx_stream_lua_shared_memory_add(ngx_conf_t *cf, ngx_str_t *name, size_t size,
+    void *tag)
+{
+    ngx_stream_lua_main_conf_t     *lmcf;
+    ngx_shm_zone_t                **zp;
+    ngx_shm_zone_t                 *zone;
+    ngx_stream_lua_shm_zone_ctx_t  *ctx;
+    ngx_int_t                     n;
```
Comment (agentzh) [inline src/ngx_stream_lua_api.c]: Style: variable name is not aligned up.

Comment (agentzh) [inline src/ngx_stream_lua_api.c]: Maybe you should run `ngx-releng` to check coding style issues?

**Comment by agentzh** [comment]
@rainingmaster Also, will you please rebase to the latest master? #56 has already been merged, for example :) Thanks!

File: src/ngx_stream_lua_api.c (hunk 1 above)
Comment (rainingmaster) [inline src/ngx_stream_lua_api.c]: @agentzh Sor, I forgot to check again after some batch replace operation. I have fix these issues and rebase to the lastest master~

**Comment by dndx** [comment]
This is no longer necessary as we already had https://github.com/openresty/stream-lua-nginx-module/blob/master/src/ngx_stream_lua_api.c.

**Pull request closed without merging**